common/src/main/java/com/codeforcommunity/auth/Passwords.java

```diff
@@ -0,0 +1,101 @@
+package com.codeforcommunity.auth;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Arrays;
+import java.util.Random;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+
+/**
+ * A utility class to hash passwords and check passwords vs hashed values. It uses a combination of hashing and unique
+ * salt. The algorithm used is PBKDF2WithHmacSHA1 which, although not the best for hashing password (vs. bcrypt) is
+ * still considered robust and <a href="https://security.stackexchange.com/a/6415/12614"> recommended by NIST </a>.
+ * The hashed value has 256 bits.
+ */
+public class Passwords {
```

Comment:
In the future don't waste time on classes that are outside the scope of your ticket. It just wastes time by trying to anticipate the requirements of problems we're not trying to solve yet

Reply:
This was copied from a stackoverflow answer and modified a little. Just thought I'd throw it in just in case because it has a few practices I'd like to see used.